src/etc/inc/interfaces.inc

```diff
@@ -1385,6 +1321,125 @@ function interface_gif_configure(&$gif, $gifkey = "") {
 	return $gifif;
 }
 
+/* NOTE: $vxlankey is not used but useful for passing this function to array_walk. */
+function interface_vxlan_configure(&$vxlan, $vxlankey = "") {
+	global $config, $g;
+
+	if (!is_array($vxlan)) {
+		return -1;
+	}
+
+	$realif = convert_friendly_interface_to_real_interface_name($vxlan['if']);
+	if (!interface_is_vlan($realif)) {
+		$realif = get_real_interface($vxlan['if']);
+	}
+	$realifip = get_interface_ip($vxlan['if']);
+	$realifip6 = get_interface_ipv6($vxlan['if']);
+
+	/* make sure the parent interface is up */
+	if ($realif) {
+		interfaces_bring_up($realif);
+	} else {
+		log_error(gettext("could not bring parentif up -- variable not defined -- interface_vxlan_configure()"));
+	}
+
+	if (!platform_booting() && !(empty($vxlan['vxlanif']))) {
+		pfSense_interface_destroy($vxlan['vxlanif']);
+		pfSense_interface_create($vxlan['vxlanif']);
+		$vxlanif = $vxlan['vxlanif'];
+	} else {
+		$vxlanif = pfSense_interface_create("vxlan");
+	}
+
+	$ifconfigcmd = "/sbin/ifconfig " . escapeshellarg($vxlanif) . " vxlanid " .
+		escapeshellarg($vxlan['vxlanid']) . " vxlandev " . escapeshellarg($realif);
```

Comment:
Multiline statements must use 4 spaces to indent

Reply:
ok, fixed!